### PR TITLE
Update generic messenger doc

### PR DESCRIPTION
**Pull request by mooreds**
Per comments in this issue: https://github.com/FusionAuth/fusionauth-issues/issues/1566 I need to update the generic messenger doc to reflect how to actually use a generic messenger.

### CODE REVIEW

File: site/docs/v1/tech/messengers/generic-messenger.adoc [hunk 1]
```diff
@@ -80,3 +80,43 @@ In these fields, you can add custom key and value pairs to augment your headers.
 image::messengers/configuration-test-generic.png[Test Configuration Generic, role=top-cropped, width=1200]
 
 You also can test your generic messenger configuration. By hitting `Test generic configuration` FusionAuth will fire a JSON message to the messenger just created to ensure everything is set up correctly.
+
+== Writing the Generic Message Receiver
+
+You need to write a server side component, a receiver, which will be deployed at a URL and send the actual messages. This component can be written in any language that can consume a JSON message.
```
Comment (robotdan) [inline site/docs/v1/tech/messengers/generic-messenger.adoc:86]: > deployed at a URL and send the actual messages. 

This almost sounds like the "receiver" is sending the messages. What if we said something like:

> deployed at a URL to consume the message sent from FusionAuth, and proxy the request to the 3rd party messaging service.

File: site/docs/v1/tech/messengers/generic-messenger.adoc [hunk 2]
```diff
@@ -80,3 +80,43 @@ In these fields, you can add custom key and value pairs to augment your headers.
...
+.Example Request JSON
+----
+include::docs/src/json/messengers/generic-message-request.json[]
+----
+
+The components of this message include:
+
+* [field]#phoneNumber#, the number of the user to which to send the message.
+* [field]#textMessage#, the message to send. It is built from the configured link:/https://fusionauth.io/docs/v1/tech/email-templates/message-templates/[message template] and will be localized.
+* [field]#type#, which is always `SMS`.
```
Comment (robotdan) [inline site/docs/v1/tech/messengers/generic-messenger.adoc:102]: Can we structure this just like any other API doc with field definitions?

File: site/docs/v1/tech/messengers/generic-messenger.adoc [hunk 3]
```diff
@@ -80,3 +80,43 @@ In these fields, you can add custom key and value pairs to augment your headers.
...
+
+When your application receives this message, it should send the notification to the phone number provided.
```
Comment (robotdan) [inline site/docs/v1/tech/messengers/generic-messenger.adoc:104]: Or proxy to the 3rd party that is actually going to send the message? I would assume it is unlikely the builder of this proxy/component will be the one sending the message. 

The proxy could call another SDK to send the message, or build another API request to a 3rd party.

**Review by robotdan — APPROVED** [review]

File: site/docs/src/json/messengers/generic-message-request.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "phoneNumber": "3035551212",
+  "textMessage": "Two Factor Code: 111111",
+  "type": "SMS"
```
Comment (robotdan) [inline site/docs/src/json/messengers/generic-message-request.json:4]: Just FYI, reviewing some of this code as the result of this doc update. Opened this GH issue to add a few more things to make integration easier. 
https://github.com/FusionAuth/fusionauth-issues/issues/1574

File: site/docs/v1/tech/messengers/generic-messenger.adoc [hunk 5]
```diff
@@ -11,17 +11,17 @@ Available since 1.26.0
 
 == Set Up a Generic Messenger
 
-Currently, messengers can only be configured to send messages via a SMS protocol. If you need to connect to an SMS provider other than Twilio, you can use the Generic messenger configuration.
+Currently, built-in messengers can only be configured to send messages via a Twilio. If you need to connect to a provider other than Twilio, whether you want to send SMS messages or any other, you can use the Generic messenger configuration.
 
-The generic messenger allows for a simple JSON REST API integration which allows you to receive a message from FusionAuth and then make an API call to any third party SMS provider you wish.
+The generic messenger allows for a simple JSON REST API integration which allows you to receive a message from FusionAuth to an arbitrary URL. The Generic Message Receiver at this URL may then make an API call to any third party messaging provider.
```
Comment (jobannon) [inline site/docs/v1/tech/messengers/generic-messenger.adoc:16]: I am not sure if this is needed, but a thought:

> In other words, a Generic Messenger is a thin coordination layer between FusionAuth and other messaging services (with the proper authority and functionality to act on a message).

File: site/docs/v1/tech/messengers/generic-messenger.adoc (hunk 2 above)
Comment (mooreds) [reply site/docs/v1/tech/messengers/generic-messenger.adoc]: Sure.

**Review by jobannon — APPROVED** [review]
Looks good to me!

File: site/docs/v1/tech/messengers/generic-messenger.adoc (hunk 5 above)
Comment (mooreds) [reply site/docs/v1/tech/messengers/generic-messenger.adoc:16]: Thanks, I tweaked it a bit, but I like the 'thin coordination layer' verbiage.

**Pull request merged**